src/op_mode/generate_ipsec_debug_archive.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2022 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from datetime import datetime
+from pathlib import Path
+from shutil import rmtree
+from socket import gethostname
+from sys import exit
+from tarfile import open as tar_open
+from vyos.util import rc_cmd
+
+# define a list of commands that needs to be executed
+CMD_LIST: list[str] = [
+    'sudo ipsec status',
+    'sudo swanctl -L',
+    'sudo swanctl -l',
+    'sudo swanctl -P',
+    'sudo ip x sa show',
+    'sudo ip x policy show',
+    'sudo ip tunnel show',
+    'sudo ip address',
+    'sudo ip rule show',
+    'sudo ip route | head -100',
+    'sudo ip route show table 220'
+]
+JOURNALCTL_CMD: str = 'sudo journalctl -b -n 10000 /usr/lib/ipsec/charon'
+
+
+# execute a command and save the output to a file
+def save_stdout(command: str, file: Path) -> None:
+    rc, stdout = rc_cmd(command)
+    body: str = f'''###
+Command: {command}
+Exit code: {rc}
+Stdout:
+{stdout}
+'''
```

Comment:
Add please a new line after {stdout}
In 'sh' script it was:
```
### sudo ip rule show ###
0:	from all lookup local
32766:	from all lookup main
32767:	from all lookup default

### sudo ip route ###
default nhid 13 via 192.168.122.1 dev eth0 proto static metric 20 
192.0.2.0/24 dev eth1 proto kernel scope link src 192.0.2.1 
192.168.122.0/24 dev eth0 proto kernel scope link src 192.168.122.14 

### sudo ip route show table 220 ###
root@r14:/tmp# 
```
Wich your change:
```
###
Command: sudo ip rule show
Exit code: 0
Stdout:
0:	from all lookup local
32766:	from all lookup main
32767:	from all lookup default
###
Command: sudo ip route | head -100
Exit code: 0
Stdout:
default nhid 13 via 192.168.122.1 dev eth0 proto static metric 20 
192.0.2.0/24 dev eth1 proto kernel scope link src 192.0.2.1 
192.168.122.0/24 dev eth0 proto kernel scope link src 192.168.122.14
###
Command: sudo ip route show table 220
Exit code: 2
Stdout:
Error: ipv4: FIB table does not exist.
Dump terminated
root@r14:/tmp# 
```

Reply:
I ve chanched the code and it's more readable :
```

def save_stdout(command: str, file: Path) -> None:
    rc, stdout = rc_cmd(command)
    body: str = f'''### {command} ###
Command: {command}
Exit code: {rc}

Stdout:
{stdout}

'''


```